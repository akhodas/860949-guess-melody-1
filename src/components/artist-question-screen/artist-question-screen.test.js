import React from 'react';
import renderer from 'react-test-renderer';

import ArtistQuestionScreen from './artist-question-screen.jsx';

describe(`ArtistQuestionScreen`, () => {
  it(`correctly rendered`, () => {
    const tree = renderer
      .create(<ArtistQuestionScreen
        question={{
          type: `artist`,
          song: {
            artist: `artist1`,
            src: `path`,
          },
          answers: [
            {
              picture: `picture0`,
              artist: `artist0`,
            },
            {
              picture: `picture1`,
              artist: `artist1`,
            },
            {
              picture: `picture2`,
              artist: `artist2`,
            },
          ],
        }}
        onAnswer={jest.fn()}
      />, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
