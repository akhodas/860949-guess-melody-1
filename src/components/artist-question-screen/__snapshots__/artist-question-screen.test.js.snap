// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen correctly rendered 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
    <div
      className="track__status"
    >
      <audio />
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-0"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="artist-0"
      >
        <img
          alt="artist0"
          className="artist__picture"
          src="picture0"
        />
        artist0
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-1"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="artist-1"
      >
        <img
          alt="artist1"
          className="artist__picture"
          src="picture1"
        />
        artist1
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-2"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="artist-2"
      >
        <img
          alt="artist2"
          className="artist__picture"
          src="picture2"
        />
        artist2
      </label>
    </div>
  </form>
</section>
`;
